driver/src/price_finding/linear_optimization_price_finder.rs

```diff
@@ -168,3 +178,275 @@ fn read_output() -> std::io::Result<serde_json::Value> {
+        let state = models::State {
+            state_hash: "hash".to_string(),
+            state_index:  0,
+            balances: vec![],
+        };
+        let return_result = || { Ok(json!({
+            "prices": {
+                "token0": "14024052566155238000",
+                "token1": "1526784674855762300",
+            },
+            "orders": []
+        }))};
+        let mut solver = LinearOptimisationPriceFinder {
+            num_tokens: 2,
+            previous_prices: HashMap::new(),
+            write_input: |_, _| Ok(()),
+            run_solver: |_| Ok(()),
+            read_output: return_result,
+        };
+
+        solver.find_prices(&vec![], &state).expect("Should not fail");
+
+        let expected_prices: Prices = [
+            ("token0".to_owned(), "14024052566155238000".to_owned()),
+            ("token1".to_owned(), "1526784674855762300".to_owned())
+        ].iter().cloned().collect();
+
+        assert_eq!(solver.previous_prices, expected_prices);
+    }
+
+    #[test]
+    fn test_serialize_order() {
+        let order = models::Order {
+            account_id: 1,
+            sell_token: 2,
+            buy_token: 3,
+            sell_amount: 100,
+            buy_amount: 200,
+        };
+        let result = serialize_order(&order, "some id");
+        let expected = json!({
+            "sellToken": "token2",
+            "buyToken": "token3",
+            "sellAmount": "100",
+            "buyAmount": "200",
+            "accountID": "1",
+            "ID": "some id"
+        });
+        assert_eq!(result, expected);
+    }
+
+    #[test]
+    fn test_serialize_balances() {
+        let balances = vec![100, 200, 300, 400, 500, 600];
+        let result = serialize_balances(&balances, 3);
+        let expected = json!({
+            "0": {
+                "token0": "100",
+                "token1": "200",
+                "token2": "300",
+            }, 
+            "1": {
+                "token0": "400",
+                "token1": "500",
+                "token2": "600",
+            }
+        });
+        assert_eq!(result, expected)
+    }
+
+    #[test]
+    #[should_panic]
+    fn test_serialize_balances_with_bad_balance_length() {
+        let balances = vec![100, 200];
+        serialize_balances(&balances, 3);
+    }
+
+    #[test]
+    fn test_deserialize_result() {
+        let json = json!({
+            "prices": {
+                "token0": "14024052566155238000",
+                "token1": "1526784674855762300",
+            },
+            "orders": [
+                {
+                    "execSellAmount": "0",
+                    "execBuyAmount": "0",
+                    "execSurplus": "0"
+                },
+                {
+                    "execSellAmount": "318390084925498118944",
+                    "execBuyAmount": "95042777139162480000",
+                    "execSurplus": "15854632034944469292777429010439194350"
```

Comment:
Do all of these long numbers have any significance? They all appear to be pretty random. Would is be less distracting to use numbers like `100...00`, `200...00`?

Reply:
Yes, good point.